### DISCUSSION TITLE
Looks like `nvim .` doesn't open the tree anymore?

### DISCUSSION THREAD

**Discussion by huyvohcmc** [nvim-tree/nvim-tree.lua#1660] [General]
I don't know if this is a bug or an expected behavior of nvim-tree now?

**Comment by pleshevskiy** [comment]
I ran into the same problem when I updated the plugin, so for this reason I stayed with the old version, where it still worked (rev: 1cf428ae676fd0f37ed3acff6f36a934521493ca)

**Reply by alex-courtis** [reply]
That is functionality that should work, depending on your configuration options such as `open_on_setup`, `ignore_buffer_on_setup`.

There are known, unavoidable race issues with netrw on startup. Please try and disable netrw completely via the globals. See `:help nvim-tree.disable_netrw`.

If that does not work, please create a [bug report](https://github.com/nvim-tree/nvim-tree.lua/issues/new?assignees=&labels=bug&template=bug_report.yml) with your configuration and steps to reproduce.

**Reply by alex-courtis** [reply]
See #1661

**Reply by huyvohcmc** [reply]
Is there any way to not show the tree with `nvim`, but open the tree with `nvim .`?

**Reply by kyazdani42** [reply]
@alex-courtis that might be a regression since `nvim .` considers `.` as "the current directory" which should open the tree if `hijack_directories.enable` is true.

**Reply by alex-courtis** [reply]
> @alex-courtis that might be a regression since `nvim .` considers `.` as "the current directory" which should open the tree if `hijack_directories.enable` is true.

Yes. c5536db removed the `hijack_directories` check on startup.

However, that doesn't reliably work when netrw is loaded, due to startup race conditions.

It will reliably work with default setup at `7282f7d` with
```lua
vim.g.loaded_netrw = 1
vim.g.loaded_netrwPlugin = 1
```

Raised #1664

**Reply by alex-courtis** [reply]
> Is there any way to not show the tree with `nvim`, but open the tree with `nvim .`?

This should be functional with default setup and netrw disabled following #1664 fix.

**Reply by alex-courtis** [reply]
@huyvohcmc please test fix #1665

```sh
cd /path/to/nvim-tree.lua
git pull
git checkout 1664-hijack_directories-not-respected-when-not-open_on_setup
```

You may see netrw instead of nvim-tree. Disable netrw as per `:help nvim-tree.disable_netrw`

**Reply by huyvohcmc** [reply]
@alex-courtis yep it's working on that branch.

**Reply by alex-courtis** [reply]
Thank you @huyvohcmc 
Did you need to disable netrw?

**Reply by huyvohcmc** [reply]
Yes, but my netrw is always disabled.

**Comment by alex-courtis** [comment]
All recent changes to startup behaviour have been reverted due to many regressions: #1668

A new mechanism to control startup behaviour is under way: #1669
